lib/reporters/default.js

```diff
@@ -0,0 +1,1125 @@
+			},
+			'line-number-selection-error': () => {
+				if (!this.isVerbose) {
+					this.lineNumberErrors.push(event);
+				}
+
+				write(
+					colors.information(
+						`${figures.warning} Could not parse ${this.relativeFile(
+							event.testFile
+						)} for line number selection`
+					)
+				);
+			},
+			'missing-ava-import': () => {
+				this.filesWithMissingAvaImports.add(event.testFile);
+
+				write(
+					colors.error(
+						`${figures.cross} No tests found in ${this.relativeFile(
+							event.testFile
+						)}, make sure to import "ava" at the top of your test file`
+					)
+				);
+			},
+			'hook-finished': () => {
+				if (!this.isVerbose || event.logs.length === 0) {
+					return;
+				}
+
+				this.lineWriter.writeLine(
+					`  ${this.prefixTitle(event.testFile, event.title)}`
+				);
+				this.writeLogs(event);
+			},
+			'selected-test': () => {
+				if (!this.isVerbose) {
+					return;
+				}
+
+				if (event.skip) {
+					this.lineWriter.writeLine(
+						colors.skip(`- ${this.prefixTitle(event.testFile, event.title)}`)
+					);
+				} else if (event.todo) {
+					this.lineWriter.writeLine(
+						colors.todo(`- ${this.prefixTitle(event.testFile, event.title)}`)
+					);
+				}
+			},
+			'uncaught-exception': () => {
+				if (!this.isVerbose) {
+					this.uncaughtExceptions.push(event);
+					return;
+				}
+
+				this.lineWriter.ensureEmptyLine();
+				this.lineWriter.writeLine(
+					colors.title(
+						`Uncaught exception in ${this.relativeFile(event.testFile)}`
+					)
+				);
+				this.lineWriter.writeLine();
+				this.writeErr(event);
+			},
+			'unhandled-rejection': () => {
+				if (!this.isVerbose) {
+					this.unhandledRejections.push(event);
+					return;
+				}
+
+				this.lineWriter.ensureEmptyLine();
+				this.lineWriter.writeLine(
+					colors.title(
+						`Unhandled rejection in ${this.relativeFile(event.testFile)}`
+					)
+				);
+				this.lineWriter.writeLine();
+				this.writeErr(event);
+			},
+			'worker-failed': () => {
+				if (!this.isVerbose) {
+					if (fileStats.declaredTests === 0) {
+						this.filesWithoutDeclaredTests.add(event.testFile);
+					}
+
+					return;
+				}
+
+				if (!this.filesWithMissingAvaImports.has(event.testFile)) {
+					if (event.nonZeroExitCode) {
+						this.lineWriter.writeLine(
+							colors.error(
+								`${figures.cross} ${this.relativeFile(
+									event.testFile
+								)} exited with a non-zero exit code: ${event.nonZeroExitCode}`
+							)
+						);
+					} else {
+						this.lineWriter.writeLine(
+							colors.error(
+								`${figures.cross} ${this.relativeFile(
+									event.testFile
+								)} exited due to ${event.signal}`
+							)
+						);
+					}
+				}
+			},
+			'worker-finished': () => {
+				if (
+					!event.forcedExit &&
+					!this.filesWithMissingAvaImports.has(event.testFile)
+				) {
+					if (fileStats.declaredTests === 0) {
+						if (!this.isVerbose) {
+							this.filesWithoutDeclaredTests.add(event.testFile);
+						}
```

Comment:
Here and elsewhere, I don't think there's any downside to recording this even when verbose. Let's simplify the code.